### PR TITLE
Enable automatic redirect for outdated pages in the docs.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
For the moment, I've only enabled it for the page "language basics". If the PR is merge, it should be trivial to add other pages.

The behavior is checking for the `html` build, then wait until it's finished, then put an `html` page with the redirection directly into the build directory.

I think it's important to make true redirects rather than just putting a reference in the `.rst` file because:

* We can remove `.rst` files from the documentation, making it easier to manage.
* We can remove all outdated links and `.html` pages making it easier for users to navigate in the documentation.
* URLs which get redirected are automatically detected by modern tools from other website (for example we have **linkcheck** in Cython for the documentation which does it). That enables them to automatically update the URL. Without this, it's up to the web developer to notice it and  update it manually (in practice I doubt it ever happens).

### CODE REVIEW

**Comment by scoder** [comment]
One issue: redirecting to an exact URL would discard any anchor references from the incoming URL. When I go to "`reference/compilation.html#pyximport`", it would redirect me to "`userguide/compilation.html`", and I wouldn't find what I'm looking for. Could be fixed with a bit of javascript in the redirect page, I'm sure.

File: docs/conf.py [hunk 1]
```diff
@@ -462,12 +462,16 @@
 html_template = """
 <html>
     <head>
-        <meta http-equiv="refresh" content="1; url={0}:" />
```
Comment (scoder) [inline docs/conf.py]: I'd keep this as a fallback for cases where scripts are disabled.

File: docs/conf.py [hunk 2]
```diff
@@ -462,12 +462,16 @@
...
         <script>
-            window.location.href = "{0}"
+            // Some browsers include the hash character in the anchor, strip it out
+            const anchor = window.location.hash.replace(/^#(.*)/, '$1');
+
+            // Redirect to the new-style URL
+            window.location = '{}#' + anchor;
```
Comment (scoder) [inline docs/conf.py]: This would append a useless `#` to the URL if anchor is empty.

**Comment by gabrieldemarmiesse** [comment]
Yay! I'm really happy that this one is merged!

**Comment by gabrieldemarmiesse** [comment]
Mmhh not cool. I'm getting a "too many redirect". I'll see what I can do.

**Comment by gabrieldemarmiesse** [comment]
It seems that the HTML page (the one that is supposed to redirect) just doesn't exist. Could it be linked to #2180 ? Would it be possible to have access to a build log or something from Readthedocs?

**Comment by scoder** [comment]
Where do you get the redirect? The TOC at http://docs.cython.org/ contains only a single entry of "language basics", and that one opens correctly for me.

**Comment by gabrieldemarmiesse** [comment]
https://cython.readthedocs.io/en/latest/src/reference/language_basics.html

**Comment by scoder** [comment]
I have no idea why RTD does that – they send a redirect for every non-existant URL that I tried, but send me back the exact same URL again.
Anyway, that suggests that the HTML redirect page was not generated. Also no idea why.

**Comment by gabrieldemarmiesse** [comment]
I reached the same conclusion, and in the end, I don't understand why, for example autodocs would work, but not the small function that I made. Well, if you come across any hint, let me know.

**Comment by gabrieldemarmiesse** [comment]
Maybe it would be a good idea to undo this PR then until we find another way?

**Pull request merged**